### PR TITLE
Bugfix/496 rollover alias with filters

### PR DESCRIPTION
**Pull request by stevanbz**
### Description
Once the rollover happens on alias, filter, search and index routing both on source and target indicies were missing.
This pull request should fix the previously mentioned behaviors by remembering the alias filter, search and index routing settings. 

### Issues Resolved
https://github.com/opensearch-project/index-management/issues/496

### Check List

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
Thanks @stevanbz, please amend with `-s` for DCO.

**Review by dblock — APPROVED** [review]
Looks pretty straightforward, thanks.

**Comment by stevanbz** [comment]
> Thanks @stevanbz, please amend with `-s` for DCO.

Tnx for reviewing the PR. Tnx for letting me know.
Will do so

File: CHANGELOG.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Getting security exception due to access denied 'java.lang.RuntimePermission' 'accessDeclaredMembers' when trying to get snapshot with S3 IRSA ([#4469](https://github.com/opensearch-project/OpenSearch/pull/4469))
 - Fixed flaky test `ResourceAwareTasksTests.testTaskIdPersistsInThreadContext` ([#4484](https://github.com/opensearch-project/OpenSearch/pull/4484))
 - Fixed the ignore_malformed setting to also ignore objects ([#4494](https://github.com/opensearch-project/OpenSearch/pull/4494))
-
```
Comment (dblock) [inline CHANGELOG.md:68]: Leave the extra line please?

**Review by dblock — APPROVED** [review]

File: server/src/main/java/org/opensearch/cluster/metadata/AliasMetadata.java [hunk 2]
```diff
@@ -129,6 +129,13 @@ public CompressedXContent getFilter() {
         return filter();
     }
 
+    public String getFilterAsString() {
```
Comment (reta) [inline server/src/main/java/org/opensearch/cluster/metadata/AliasMetadata.java:132]: I am wondering it adding this method is really needed (the `filter()` is exposed already), the consumer could do that transformation easily

File: server/src/main/java/org/opensearch/action/admin/indices/rollover/MetadataRolloverService.java [hunk 3]
```diff
@@ -309,20 +309,46 @@ static List<AliasAction> rolloverAliasToNewIndex(
         String oldIndex,
         String newIndex,
         boolean explicitWriteIndex,
-        @Nullable Boolean isHidden,
+        AliasMetadata aliasMetada,
```
Comment (andrross) [inline server/src/main/java/org/opensearch/action/admin/indices/rollover/MetadataRolloverService.java:312]: Typo here on "aliasMetada", should be "aliasMetadata"

**Review by andrross — COMMENTED** [review]
Looks good overall, just found one typo

File: server/src/main/java/org/opensearch/action/admin/indices/rollover/MetadataRolloverService.java (hunk 3 above)
Comment (stevanbz) [reply server/src/main/java/org/opensearch/action/admin/indices/rollover/MetadataRolloverService.java:312]: @andrross good catch! Tnx!

**Review by andrross — APPROVED** [review]

**Comment by turesheim** [comment]
Any chance we can get this backported to 1.3.x? I'd be happy to do the work.

**Pull request merged**